### DISCUSSION TITLE
How to compare CryoDRGN with CryoSPARC

### DISCUSSION THREAD

**Discussion by qhz991029** [zhonge/cryodrgn#364] [Q&A] [answered]
Thank you for your great work!

I am now trying to learn CryoDRGN and tried to reproduce the FSC result on EMPIAR-10028 between CryoDRGN and CryoSPARC. I followed the description in CryoDRGN document (first ab inito then homogeneous refinement) to reconsturct EMPIAR-10028 using CryoSPARC. Then I trained CryoDRGN using all particle images at resolution 360*360 for 50 epochs (256 dim * 3 layers) with the pose and ctf provided by your work in another repo. I tried to calculate FSC after alignment using CryoSPARC but the curve turned out to be very bad.
![cryodrgn_cryosparc](https://github.com/ml-struct-bio/cryodrgn/assets/90995734/bd27f68e-7963-40f2-b4c6-cd4738037e43)
Here are the mrc files I used:

- My CryoDRGN reconstruction: https://drive.google.com/file/d/16A0j67H9ivQt07eBiZDuepT-tRiWdOLD/view?usp=sharing
- My CryoSPARC reconstruction: https://drive.google.com/file/d/16FVbf6yq7fL-kwA31wzra5Wkbek3lUej/view?usp=sharing

I wonder what would be the potential reason. Is there anything I need to adjust when reconstruct using CryoSPARC? I appreciate any feedback.

Best ragards.

**Comment by michal-g** [comment]
Hi, can you provide a little more information on the settings you used in your CRYOSPARC reconstruction, and if possible, the `config.yaml` file output as part of cryoDRGN training? But yes, this can be caused by any number of possible discrepancies between how the cryoDRGN and CRYOSPARC reconstructions were exactly done.

**Comment by qhz991029** [comment]
Thank you for your reply!

For the CryoSPARC v4.4.1 reconstruction, I used the [official 50.8 GB particle images](https://www.ebi.ac.uk/empiar/EMPIAR-10028/). I first imported the particle stack, then performed an ab initio reconsturction, and then a homogeneous refinement. All the parameters are set to default as described in the CryoDRGN document.

This is my CryoDRGN training config: https://drive.google.com/file/d/16KdlKEcH15oDe8AcufJwhBl2pYR1T3b-/view?usp=sharing

Besides I also calculated FSC between my CryoDRGN recon. and the [reported EMPIAR-10028 density map](https://www.emdataresource.org/EMD-2660), and the result is very close to the number reported in the paper.
![cryodrgn_gt](https://github.com/ml-struct-bio/cryodrgn/assets/90995734/4a8c10fd-ab7f-4dba-9ca4-1f394a2fa2ab)

These two FSC figures are automatically calculated using CryoSPARC align 3D tool because I find all the density maps are not aligned.

**Comment by michal-g** [comment]
According to your `config.yaml`, you performed a reconstruction with known poses (in this case, `/empiar-10028/poses.pkl`), while your CRYOSPARC job was ab initio (no known poses used). Give it another shot with `cryodrgn abinit_homo`!

**Comment by zhonge** [comment, marked as answer]
I took a brief look at your shared .mrc files and the volumes look good, so I'm not sure why the FSC curve is so bad. Can you check if the volumes were appropriately aligned before computing the FSC?

If you're interested in an apples-to-apples comparison between cryoSPARC and cryoDRGN with the same poses, I would use `cryodrgn train_nn`, but if you also want to do ab initio reconstruction in cryoDRGN, you can use `cryodrgn abinit_homo` (more experimental).

I would also recommend downsampling the images from D=360 or D=256 or even D=128 depending on what you're trying to do, since it will be much faster to train!

**Reply by qhz991029** [reply]
Thank you so much for your answer, after using `train_nn`, my CryoDRGN reconstruction is in the same pose with CryoSPARC, and FSC could reach 4 angstrom.
